pkg/io/binaryWriter.go

```diff
@@ -93,18 +93,18 @@ func (w *BinWriter) WriteVarUint(val uint64) {
 
 }
 
-// WriteVarBytes writes a variable byte into the underlying io.Writer without prefix.
-func (w *BinWriter) WriteVarBytes(b []byte) {
-	w.WriteLE(b)
+// WriteBytes writes a variable byte into the underlying io.Writer without prefix.
+func (w *BinWriter) WriteBytes(v interface{}) {
```

Comment:
`WriteBytes(b []byte)`?

Reply:
hm  in some places we are not using []byte but we have byte or uint16 or uint32 or unit256 which used WriteLE. Do you mean we have to leave there WriteLE? In this case we would have very mixed usages there..

Reply:
> we have byte or uint16 or uint32

These are perfect candidates for `WriteLE`.

> or unit256

Although this one is not so much, it should be explicitly converted to LE or BE `[]byte` and then written with `WriteBytes`. At the moment what you're actually writing with `WriteLE` for this is a BE value.

> In this case we would have very mixed usages there..

Why? `WriteBytes` for slices of bytes (that don't and can't have any notion of LE/BE), `WriteLE/WriteBE` for integers (that do care about LE/BE a lot), simple as that.

Reply:
Ok, reworked. But `WriteBytes` used not a lot and having it as for me doesn't make much sense. Anyway here it is.